src/shared/ifdef.rs

```diff
@@ -3,20 +3,198 @@
 // <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your option.
 // All files in the project carrying such notice may not be copied, modified, or distributed
 // except according to those terms.
-use shared::basetsd::{UINT16, ULONG64};
-use shared::minwindef::ULONG;
-STRUCT!{struct NET_LUID_LH {
-    Value: ULONG64,
+use shared::basetsd::{UINT16, UINT32, ULONG32, ULONG64};
+use shared::guiddef::GUID;
+use shared::ntdef::{BOOLEAN, UCHAR, ULONG, USHORT, WCHAR};
+pub type NET_IF_COMPARTMENT_ID = UINT32;
+pub type PNET_IF_COMPARTMENT_ID = *mut NET_IF_COMPARTMENT_ID;
+pub const NET_IF_COMPARTMENT_ID_UNSPECIFIED: NET_IF_COMPARTMENT_ID = 0;
+pub const NET_IF_COMPARTMENT_ID_PRIMARY: NET_IF_COMPARTMENT_ID = 1;
+pub type NET_IF_NETWORK_GUID = GUID;
+pub type PNET_IF_NETWORK_GUID = *mut NET_IF_NETWORK_GUID;
+ENUM!{enum NET_IF_OPER_STATUS {
+    NET_IF_OPER_STATUS_UP = 1,
+    NET_IF_OPER_STATUS_DOWN = 2,
+    NET_IF_OPER_STATUS_TESTING = 3,
+    NET_IF_OPER_STATUS_UNKNOWN = 4,
+    NET_IF_OPER_STATUS_DORMANT = 5,
+    NET_IF_OPER_STATUS_NOT_PRESENT = 6,
+    NET_IF_OPER_STATUS_LOWER_LAYER_DOWN = 7,
 }}
-BITFIELD!{NET_LUID_LH Value: ULONG64 [
+pub type PNET_IF_OPER_STATUS = *mut NET_IF_OPER_STATUS;
+pub type NET_IF_OBJECT_ID = ULONG32;
+pub type PNET_IF_OBJECT_ID = *mut NET_IF_OBJECT_ID;
+ENUM!{enum NET_IF_ADMIN_STATUS {
+    NET_IF_ADMIN_STATUS_UP = 1,
+    NET_IF_ADMIN_STATUS_DOWN = 2,
+    NET_IF_ADMIN_STATUS_TESTING = 3,
+}}
+pub type PNET_IF_ADMIN_STATUS = *mut NET_IF_ADMIN_STATUS;
+pub type NET_IF_COMPARTMENT_SCOPE = UINT32;
+pub type PNET_IF_COMPARTMENT_SCOPE = *mut NET_IF_COMPARTMENT_SCOPE;
+pub const NET_IF_COMPARTMENT_SCOPE_UNSPECIFIED: NET_IF_COMPARTMENT_SCOPE = 0;
+pub const NET_IF_COMPARTMENT_SCOPE_ALL: NET_IF_COMPARTMENT_SCOPE = -1i32 as
+    NET_IF_COMPARTMENT_SCOPE;
```

Comment:
```suggestion
pub const NET_IF_COMPARTMENT_SCOPE_ALL: NET_IF_COMPARTMENT_SCOPE = -1i32 as u32;
```
As per https://github.com/retep998/winapi-rs/blob/0.3/CONTRIBUTING.md#constants